sql/src/main/java/io/crate/planner/node/dql/AbstractDQLPlanNode.java

```diff
@@ -58,10 +58,22 @@ public boolean hasProjections() {
         return projections != null && projections.size() > 0;
     }
 
+    @Override
     public List<Projection> projections() {
         return projections;
     }
 
+    public void projections(List<Projection> projections) {
+        this.projections = projections;
+    }
+
+    @Override
+    public void addProjection(Projection projection) {
+        List<Projection> projections = new ArrayList<>(this.projections);
```

Comment:
maybe re-create a ImmutableList so that projections().add() wouldn't work. Otherwise one could by-pass the addProjection() method.